File: FRONT/AmusementPark/src/app/models/parks/park-item.ts
import { LocalizedItem } from '../shared/localized-item';
import { AttractionDetails } from './attraction-details';
import { AttractionLocations } from './attraction-locations';
import { AdminReviewStatus } from '@app/models/admin/admin-review-status';
import { ParkItemCategory } from './park-item-category';
import { ParkItemType } from './park-item-type';

export interface ParkItem {
  id?: string;
  parkId: string;
  zoneId?: string | null;
  name: string;
  category: ParkItemCategory;
  type: ParkItemType;
  subtype?: string | null;
  latitude: number;
  longitude: number;
  descriptions?: LocalizedItem<string>[];
  attractionDetails?: AttractionDetails | null;
  attractionLocations?: AttractionLocations | null;
  isVisible?: boolean;
  adminReviewStatus?: AdminReviewStatus;
}

File: API/Entities/Entities/Model/Parks/ParkItem.cs
using Common.General;
using Common.General.Localization;
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace Entities.Model.Parks
{
    public class ParkItem : GeolocatedEntity
    {
        [BsonElement("parkId")]
        public string ParkId { get; set; } = string.Empty;

        [BsonElement("zoneId")]
        [BsonIgnoreIfNull]
        [BsonRepresentation(BsonType.String)]
        public string? ZoneId { get; set; }

        [BsonElement("name")]
        public string Name { get; set; } = string.Empty;

        [BsonElement("category")]
        [BsonRepresentation(BsonType.String)]
        public ParkItemCategory Category { get; set; }

        [BsonElement("type")]
        [BsonRepresentation(BsonType.String)]
        public ParkItemType Type { get; set; }

        [BsonElement("subtype")]
        [BsonIgnoreIfNull]
        public string? Subtype { get; set; }

        [BsonElement("descriptions")]
        public List<LocalizedItem<string>> Descriptions { get; set; } = new();

        [BsonElement("isVisible")]
        public bool IsVisible { get; set; } = true;
    }
}
